modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java

```diff
@@ -2971,10 +2981,25 @@ private void registerConsumer(C consumer, boolean nowait) throws AMQException //
     public abstract void handleAddressBasedDestination(AMQDestination dest, 
                                                        boolean isConsumer,
                                                        boolean noWait) throws AMQException;
-    
-    private void registerProducer(long producerId, MessageProducer producer)
-    {
-        _producers.put(new Long(producerId), producer);
+
+    private void registerProducer(long producerId, P producer) throws AMQException, JMSException {
+        try {
+            AMQDestination amqDestination = producer.getDestination();
+            AMQProtocolHandler protocolHandler = getProtocolHandler();
+            boolean nowait = false;
+
+            if (DECLARE_EXCHANGES) {
+                declareExchange(amqDestination, protocolHandler, nowait);
+            }
+            if (DECLARE_QUEUES || amqDestination.isNameRequired()) {
+                declareQueue(amqDestination, protocolHandler, false, nowait);
+            }
+            bindQueue(amqDestination.getAMQQueueName(), amqDestination.getRoutingKey(), FieldTableFactory
+                    .newFieldTable(), amqDestination.getExchangeName(), amqDestination, nowait);
+            _producers.put(new Long(producerId), producer);
+        } catch (JMSException e) {
+            throw e;
```

Comment:
Why are we catching the exception just to throw it. If no processing is required no point of catching.